2005-09-22  Jim Meyering  <jim at meyering.net>

	* quotearg.c (quotearg_n_options): Change code to be suboptimal, in
	order to avoid an unsuppressible warning from gcc on 64-bit systems.

--- gnulib/lib/quotearg.c	19 Sep 2005 17:28:14 -0000	1.46
+++ gnulib/lib/quotearg.c	23 Sep 2005 04:15:13 -0000	1.47
@@ -586,7 +586,12 @@ quotearg_n_options (int n, char const *a
 
   if (nslots <= n0)
     {
-      unsigned int n1 = n0 + 1;
+      /* FIXME: technically, the type of n1 should be `unsigned int',
+	 but that evokes an unsuppressible warning from gcc-4.0.1 and
+	 older.  If gcc ever provides an option to suppress that warning,
+	 revert to the original type, so that the test in xalloc_oversized
+	 is once again performed only at compile time.  */
+      size_t n1 = n0 + 1;
 
       if (xalloc_oversized (n1, sizeof *slotvec))
 	xalloc_die ();
